wsd/ClientSession.cpp

```diff
@@ -1215,6 +1215,16 @@ bool ClientSession::_handleInput(const char *buffer, int length)
             docBroker->updateLastModifyingActivityTime();
         }
 
+        /*
+            When the UI mode is switched, tile invalidation callback is sent before the mode switch callback.
+            This causes the tiles' ui mode to conflict with the view mode.
+            We set the view mode beforehand here.
+        */
+        if (firstLine == "uno .uno:NormalMultiPaneGUI")
+            _clientSelectedMode = ViewMode::NORMAL_VIEW;
+        else if (firstLine == "uno .uno:NotesMode")
```

Comment:
do we need maybe, master page mode here?

Reply:
Ah, that may work. But there may be also other issues with master switch. I think we can check this after this one.